compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -447,7 +447,7 @@ private class ExtractDependenciesCollector extends tpd.TreeTraverser { thisTreeT
     // Avoid cycles by remembering both the types (testcase:
     // tests/run/enum-values.scala) and the symbols of named types (testcase:
     // tests/pos-java-interop/i13575) we've seen before.
-    val seen = new mutable.HashSet[Symbol | Type]
+    val seen = new mutable.HashSet[Symbol | Type](initialCapacity = 128, loadFactor = mutable.HashSet.defaultLoadFactor)
```

Comment:
128 seems a bit too excessive. If we look at the histogram of maximum sizes we see that 64 is enough to cover most cases. Even 32 we would cover 87% of cases and have an extra allocation for the 13% (larger cases).


<img width="913" alt="Screenshot 2023-07-17 at 10 08 39" src="https://github.com/lampepfl/dotty/assets/3648029/6951ba6d-9c90-4068-aee1-9532b8221022">

Reply:
@nicolasstucki Nice histogram! I wanted to build one too but gave up eventually. How did you do it?
Do your percentages consider the `loadFactor` of `75%`? Should I update the `initialCapacity` to `64`?

Reply:
```scala
  def addTypeDependency(tpe: Type)(using Context): Unit = {
    val traverser = new TypeDependencyTraverser {
      def addDependency(symbol: Symbol) = addMemberRefDependency(symbol)
    }
    traverser.traverse(tpe)
    println("seen: " + traverser.seen.size)
  }

  def addPatMatDependency(tpe: Type)(using Context): Unit = {
    val traverser = new TypeDependencyTraverser {
      def addDependency(symbol: Symbol) =
        if (!ignoreDependency(symbol) && symbol.is(Sealed)) {
          val usedName = symbol.zincMangledName
          addUsedName(usedName, UseScope.PatMatTarget)
        }
    }
    traverser.traverse(tpe)
    println("seen: " + traverser.seen.size)

  }
```

`sbt "clean; scala3-bootstrapped/compile" > sbtOutput.txt` then filter out `seen: ` from that file and import the numbers in google sheets.

Reply:
I did not consider the load factor. My percentages only take into account the size of the set. I would go for 64 if we want to minimize those allocations.